File: backend/app/schemas/search_health.py
import uuid

from pydantic import BaseModel, Field

from app.schemas.insights import InsightResponse


class SearchHealthSignal(InsightResponse):
    signal_type: str
    gentle_guidance: str


class SearchHealthResponse(BaseModel):
    generated_at: str
    workspace_id: uuid.UUID | None = None
    signals: list[SearchHealthSignal] = Field(default_factory=list)
    insufficient_data: list[str] = Field(default_factory=list)

File: frontend/src/types/searchHealth.ts
import type { Insight } from "./insights";

export interface SearchHealthSignal extends Insight {
  signal_type: string;
  gentle_guidance: string;
}

export interface SearchHealthResponse {
  generated_at: string;
  workspace_id: string | null;
  signals: SearchHealthSignal[];
  insufficient_data: string[];
}
